### PR TITLE
T90 top bar refactor

### PR DESCRIPTION
**Pull request by robertknight**
Extract top bar component into own directive

In preparation for implementing the new top-bar
design, extract it into its own directive.

No functional or visual changes in this PR

T-90

### CODE REVIEW

**Comment by nickstenning** [comment]
Needs a rebase now I've merged #2615.

File: h/static/scripts/filter/test/persona-test.js [hunk 1]
```diff
@@ -15,21 +15,15 @@ describe('persona', function () {
...
-  });
-
   it('should return the requested part', function () {
     assert.equal(filter(term), 'hacker');
-    assert.equal(filter(term, 'term'), term);
     assert.equal(filter(term, 'username'), 'hacker');
     assert.equal(filter(term, 'provider'), 'example.com');
   });
 
   it('should pass unrecognized terms as username or term', function () {
```
Comment (nickstenning) [inline h/static/scripts/filter/test/persona-test.js]: You've updated the expectations but not the description of behaviour here :)

**Comment by robertknight** [comment]
Test case descriptions updated. IMO the filters are a lot of unnecessary ceremony for something so simple but I'll come back to that another time.

**Comment by nickstenning** [comment]
Great! Looking good.

**Pull request merged**